### ISSUE TITLE
Caption formatting issues when using knitr::kable() and when using symbols in chunk name

### ISSUE THREAD

**Issue by MarcioFCMartins** [rstudio/radix#46]
I have been adding captions to my documents in radix and noticed some issues with them:

  Table captions inserted using knitr::kable are formated as standard text. Would it be possible to make it so that captions inserted this way are formated like captions from fig.cap chunk option? 

  Table/figure numbering and naming does not work if your chunk name includes a space, underscore or point.

See attachment for simple examples :

[Test_captions.txt](https://github.com/rstudio/radix/files/2464369/Test_captions.txt)

**Comment by jjallaire** [comment]
Okay, I've modified the CSS for table captions to appear as follows:

<img width="718" alt="screen shot 2018-10-10 at 9 06 19 am" src="https://user-images.githubusercontent.com/104391/46738383-09568700-cc6c-11e8-9cdc-f79aae8a0d43.png">

Note that for `kable` the caption always appears above the table by design: https://github.com/yihui/knitr/issues/1189

In terms of the other cross references, this is a known limitation of bookdown cross-references: https://bookdown.org/yihui/bookdown/figures.html

I'll add a note about that to the documentation.

**Comment by MarcioFCMartins** [comment]
kable captions are now working, but adding extra formatting via kableExtra::kable_styling still breaks it. Actually, simply loading the package breaks it unless the user sets` options(kableExtra.auto_format = FALSE) `.  This makes me believe that the issue is when the format is set to 'html'  via this function.

I have no idea how pratical it is to ensure this formatting works for all package combinations (formattable comes to mind, since it works with kableExtra - https://haozhu233.github.io/kableExtra/use_kableExtra_with_formattable.html), but since kableExtra is recommended in the documentation (https://bookdown.org/yihui/rmarkdown/r-code.html#tables) I think it's worth ensuring it works.

The caption apearing above the table is proper scientific writing behaviour (atleast in most fields and from my experience) so that makes sense and it's exactly as I expect it.

I never realized that bookdown had that issue as well - thank you for the heads up and for the super quick fix - I will simply avoid extra table formatting until everything can be worked out.

**Comment by jjallaire** [comment]
Yeah, it's difficult to run down every table formatting package and imbue all of their output with correct behavior. So far we've focused on pagedtable and plain kable. We'd welcome contributions to help support others so long as they don't make our core codebase too complicated and/or fragile.

**Comment by MarcioFCMartins** [comment]
I am still very much a begginer in github colaboration so I'm going to post my suggestion here rather than attempt a pull request - which would most likely carry some issues on my part.


I noticed that uou can actually break the caption formatting with the base `kable` function if you set format to html rather than `pandoc`. 

This is because in 'distill.html' you are setting the caption style for the pandoc table class individually. This means you need to ensure all tables have `pandoc-table` class to correctly format their caption - and this has to be done for every package. 

Wouldn't it be more efficient to set the style for the `table>caption` element? This element is used to create the captions for all tables, regardless of their class (and the same is true for images, but you might want the ability to control the styles separatly).

I tried changing line 835 -  currently `$('.pandoc-table>caption').addClass("figcaption");`. Changing it to ` 
  $('caption').parent('table').addClass("figcaption")` changes all table captions (regardless of being tagged as pandoc tables) to the same style as `figcaption`.

You could also just set style for all captions, regardless of where they are used by removing the parent. If you have no need for differently formatted captions this might be the best.

**Comment by haozhu233** [comment]
https://haozhu233.github.io/kableExtra/using_kableExtra_in_radix.html

I just added a quick example of how to use kableExtra in radix. @MarcioFCMartins I hope it helps you feel better. :) 

I really love the design of radix!

**Comment by alexb523** [comment]
o man! @jrgant thank you. I've been stuck on this issue and could not figure out why my captions were rendering `(#tab:caption)`. Only populates correctly with a `-` for spaces. No periods or underscores.

**Comment by meghandownes** [comment]
Thank you for this post and the discussion.  Changing my chunk label allowed the bookdown kable_extra rendered as PDF to correctly add table headings.  You have helped my day so much. Thanks!

**Comment by mapostolides** [comment]
How do I change things like the caption colour and size, etc? There seems to be no documentation on this

**Comment by cderv** [comment]
> How do I change things like the caption colour and size, etc? There seems to be no documentation on this

You could probably use CSS for this - the exact CSS could depend on which format of table you are using. CSS and distill is documented in https://rstudio.github.io/distill/basics.html#theming and https://rstudio.github.io/distill/website.html#custom-style and https://rstudio.github.io/distill/website.html#theming 

We don't have built in way to change this using variable, you need to use custom CSS for this. 

Hope it helps

**Comment by cashcab1221** [comment]
I have also found that a caption can not contain some other symbols. I was trying with % in my caption for a while and then found this issue.
